chain/chain/src/resharding/trie_state_resharder.rs

```diff
@@ -0,0 +1,260 @@
+use std::fmt::Debug;
+use std::sync::Arc;
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_primitives::hash::CryptoHash;
+use near_store::adapter::trie_store::TrieStoreUpdateAdapter;
+use near_store::adapter::{StoreAdapter, StoreUpdateAdapter};
+use near_store::db::TRIE_STATE_RESHARDING_STATUS_KEY;
+use near_store::metrics::trie_state_metrics;
+use near_store::{DBCol, ShardTries, StorageError};
+
+use crate::resharding::event_type::ReshardingSplitShardParams;
+use crate::types::RuntimeAdapter;
+use near_chain_configs::{MutableConfigValue, ReshardingConfig, ReshardingHandle};
+use near_chain_primitives::Error;
+use near_o11y::metrics::IntGauge;
+use near_primitives::shard_layout::ShardUId;
+
+#[derive(BorshSerialize, BorshDeserialize)]
+struct TrieStateReshardingChildStatus {
```

Comment:
nit: Can you add general comment and some comments for the fields where it makes sense?